### ISSUE TITLE
Transient filter-bar: An option to select based on starting characters

### ISSUE THREAD

**Issue by fenugrec** [lxqt/pcmanfm-qt#1566]
Hi,

**Is your feature request related to a problem? Please describe.**
I'm referring to the feature where typing some characters opens a termporary Filter bar and the file/folder list is updated in realtime.

It took me a long time to get used to this mechanism, but now I generally like it,  except for a fairly frequent scenario : I type the first few letters of a file/folder that I know is there, but the filtered list is sorted alphabetically. I almost always prefer the item that I know starts with the same characters I typed. See screenshot for better explanation:

![filter_badsort](https://user-images.githubusercontent.com/10098613/163021670-c1fe1da9-5cd6-483b-b3f4-49cba6023eec.png)

I just typed in "proj", because I want to go into that folder. But because it's not alphabetically the first, some other item gets selected. In this case I can't reword the query to hit this exact folder name because it's a subset of the other one.
Some folders are much worse, having dozens of similarly-named items.
My only solutions are to either 
- erase / adjust my filter query, if possible - sometimes I try to memorize which part of a folder name is unique...
- arrow up/down a lot
- leave the keyboard, use the mouse to select the one I want. 

I don't really like any of those .

**Describe the solution you'd like**
- don't change how the filtering works, and keep the same sorting. This is all OK.
- simply pre-select the first item whose leading characters match the query. 

**Describe eventual alternatives you've considered**
- keeping the "filter bar" (`ctrl+B`) always open is a partial workaround : it will let me jump to the first matching item, but I find it less convenient to use, and it always takes space needlessly.

**Comment by tsujan** [comment]
Filtering should never change sorting and is done based on the occurrence of the filter string *anywhere* in the names of items. It makes no difference whether you use a transient filter-bar or a permanent one.

I'm afraid any behavior that contradicted the above conditions would be a bug to fix, not a feature to add.

**Comment by tsujan** [comment]
> but the filtered list is sorted alphabetically

It isn't, necessarily. It's sorted based on right-click menu → Sorting; i.e., the original sorting.

**Comment by fenugrec** [comment]
> Filtering should never change sorting

I know, that's why I specifically said "don't change the sorting".
What I'm suggesting is, to change which item is pre-selected to match the first characters preferentially, instead of just selecting the first item in the list.

**Comment by tsujan** [comment]
Yes, *selecting* based on the first characters is acceptable but only as an option. I'm not sure if it's popular. I reopen the report and change its title. We'll see if it's worth adding a new option.

Since 1.1.0 will be released soon, the option will be in 1.2.0 (November) if popular.

**Comment by fenugrec** [comment]
Thanks.

>  I'm not sure if it's popular.

Fair point. But I'd argue there's a lot of people (like me) with a lot of "muscle memory" on Windows Explorer where typing the first few chars jumps to the first matching item. In addition, it seems more natural / intuitive to type a folder name starting at the beginning, rather than trying to memorize which subset is unique enough to type in and correctly match the desired item.

**Comment by tsujan** [comment]
I'll start to think about its code after LXQt 1.1.0 is released. It might we the first code commit after 1.1.0.

Thanks for the idea.

**Comment by tsujan** [comment]
@stefonarch, I need your opinion.

Is this feature acceptable to you as the only behavior, without any option? It means that, if you have three files with names "AC", "BC" and "CC" and if you type "c" inside the view with the transient filter bar, then "CC" will be preselected, not "AC".

I ask because, first, I dislike too many options in Preferences dialog, and, second, I found a practical reason for it: finding "CC" could be hard *without it* if there are many filtered items.

I also tested in a folder with ~68,000 item and didn't see a tangible difference in CPU usage.

If the answer is "yes", we could have it in 1.1.0 — it'll be just a small patch.

**Comment by stefonarch** [comment]
I'm +1  for that. I usually use or the first letters or the suffix for filtering. An option could be an overkill.

**Comment by tsujan** [comment]
Done in https://github.com/lxqt/pcmanfm-qt/pull/1568

**Comment by fenugrec** [comment]
Awesome, thanks for working on this so fast. Looking forward to testing.
